app_dart/lib/src/service/github_checks_service.dart

```diff
@@ -114,6 +127,20 @@ class GithubChecksService {
     return true;
   }
 
+  /// Check if we should rerun a failed build.
+  ///
+  /// A presubmit retry happens if and only if
+  ///   1) the user is an auto roller account
+  ///   2) the same check run has never been retried before
+  bool shouldRerun(push_message.BuildPushMessage buildPushMessage) {
+    final push_message.Build build = buildPushMessage.build!;
+    if (!buildPushMessage.userData.containsKey('user_login') ||
+        !config.rollerAccounts.contains(buildPushMessage.userData['user_login'])) {
+      return false;
+    }
+    return build.tagsByName('retry').isEmpty || build.tagsByName('retry').contains('false');
```

Comment:
I am not sure I follow here. Are we tracking the number of retries through tags? or are we looking at a specific tag?

Reply:
I am using the `retry` tag to track if the current build is a rerun. Put it this way considering we only want to retry once.

Reply:
Retries should be configurable based on a target property.